clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4486,11 +4486,100 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (IsIAMCU && types::isCXX(Input.getType()))
     D.Diag(diag::err_drv_clang_unsupported) << "C++ for IAMCU";
 
+  // Handle depscan.
+  if (JA.getKind() == Action::DepscanJobClass) {
+    CmdArgs.push_back("-cc1depscan");
+
+    // Pass depscan related options to cc1depscan.
+    const OptSpecifier DepScanOpts[] = {
+        options::OPT_fdepscan_EQ,
+        options::OPT_fdepscan_share_EQ,
+        options::OPT_fdepscan_share_parent,
+        options::OPT_fdepscan_share_parent_EQ,
+        options::OPT_fno_depscan_share,
+        options::OPT_fdepscan_share_stop_EQ,
+        options::OPT_fdepscan_daemon_EQ,
+        options::OPT_fdepscan_prefix_map_EQ,
+        options::OPT_fdepscan_prefix_map_sdk_EQ,
+        options::OPT_fdepscan_prefix_map_toolchain_EQ};
+
+    Args.AddAllArgs(CmdArgs, DepScanOpts);
+
+    assert(Output.isFilename() && "Depscan needs to have file output");
+    CmdArgs.push_back("-o");
+    CmdArgs.push_back(Output.getFilename());
```

Comment:
Can we use `-fdepscan-output` here so we don't collide with `-o`?

Reply:
Actually, I don't think we'd even collide. Everything from after `-cc1-args` will get processed separately, so we could have:
```
-o file.rsp --cc1-args -emit-llvm -o file.bc
```
without any trouble, right?

In which case, I can't decide whether reusing `-o` or having a separate flag `-fdepscan-output` is better.

Reply:
Yes, exactly. `--cc1-args` split the command line and everything afterwards are just argument values, not flags.